services/business/checkers/virtual_services/subset_presence_checker.go

```diff
@@ -0,0 +1,152 @@
+	DestinationRules []kubernetes.IstioObject
+	VirtualService   kubernetes.IstioObject
+}
+
+func (checker SubsetPresenceChecker) Check() ([]*models.IstioCheck, bool) {
+	valid := true
+	validations := make([]*models.IstioCheck, 0)
+
+	http := checker.VirtualService.GetSpec()["http"]
+	if http == nil {
+		return validations, valid
+	}
+
+	// Getting a []HTTPRoute
+	slice := reflect.ValueOf(http)
+	if slice.Kind() != reflect.Slice {
+		return validations, valid
+	}
+
+	for i := 0; i < slice.Len(); i++ {
+		httpRoute, ok := slice.Index(i).Interface().(map[string]interface{})
+		if !ok || httpRoute["route"] == nil {
+			continue
+		}
+
+		// Getting a []DestinationWeight
+		destinationWeights := reflect.ValueOf(httpRoute["route"])
+		if destinationWeights.Kind() != reflect.Slice {
+			return validations, valid
+		}
+
+		for j := 0; j < destinationWeights.Len(); j++ {
+			destinationWeight, ok := destinationWeights.Index(j).Interface().(map[string]interface{})
+			if !ok || destinationWeight["destination"] == nil {
+				valid = false
+				validation := models.BuildCheck("Destination field is mandatory", "error",
+					"spec/http["+strconv.Itoa(i)+"]/route["+strconv.Itoa(j)+"]")
+				validations = append(validations, &validation)
+				continue
+			}
+
+			destination, ok := destinationWeight["destination"].(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			host, ok := destination["host"].(string)
+			if !ok {
+				continue
+			}
+
+			subset, ok := destination["subset"].(string)
+			if !ok {
+				continue
+			}
+
+			// Are there pods for that host and subset
+			subsetSelector := checker.getSelector(host, subset)
+			if subsetSelector == nil {
+				valid = false
+				validation := models.BuildCheck("Subset not found", "warning",
+					"spec/http["+strconv.Itoa(i)+"]/route["+strconv.Itoa(j)+"]/destination")
+				validations = append(validations, &validation)
+			}
+		}
+	}
+
+	return validations, valid
+}
+
+func (checker SubsetPresenceChecker) getSelector(host string, subset string) labels.Selector {
+	destinationRule, ok := checker.getDestinationRule(host)
+	if !ok {
+		return nil
+	}
+
+	labels, ok := GetSubsetLabel(destinationRule, subset)
+	if !ok {
+		return nil
+	}
+
+	return labels
+}
+
+func (checker SubsetPresenceChecker) getDestinationRule(virtualServiceHost string) (kubernetes.IstioObject, bool) {
+	for _, destinationRule := range checker.DestinationRules {
+		host, ok := destinationRule.GetSpec()["host"]
+		if !ok {
+			continue
+		}
+
+		sHost, ok := host.(string)
+		if !ok {
+			continue
+		}
+
+		domainParts := strings.Split(sHost, ".")
+		serviceName := domainParts[0]
+		namespace := checker.Namespace
+		if len(domainParts) > 1 {
+			namespace = domainParts[1]
+		}
+
+		if kubernetes.CheckHostnameService(virtualServiceHost, serviceName, namespace) {
+			return destinationRule, true
+		}
+	}
+
+	return nil, false
+}
+
+func GetSubsetLabel(destinationRule kubernetes.IstioObject, subsetTarget string) (labels.Selector, bool) {
```

Comment:
I found confusing using the labels.Selector here.
Subset labels are just a plain map[string]string and parsing to the labels perhaps is not necessary.

Reply:
This was mainly for the version previous of your first messages. Now, I changed the signature of the method according to the meaning. It returns a boolean that says if the subset is present or not.